cmd/flux/push_artifact.go

```diff
@@ -101,8 +129,39 @@ func pushArtifactCmdRun(cmd *cobra.Command, args []string) error {
 	ctx, cancel := context.WithTimeout(context.Background(), rootArgs.timeout)
 	defer cancel()
 
+	var ociClientOpts []crane.Option
+
+	if pushArtifactArgs.creds != "" {
```

Comment:
Hey @adrien-f , Thanks for this pull request🎉!

I think that most of this logic should be in `pkg/oci` so that it can be reused across the other oci commands, maybe pass the token/auth options to `NewLoclaClient` instead. The tests for this would also be added there.
I think the `--auto-login` thing is pretty neat

cc: @stefanprodan

Reply:
Yes besides `flux push` there are other commands which make use of the OCI client. I guess @adrien-f didn't run `make build` or `make vet` or `make test` all of those would've fail. Instead of `--auto-login` I would prefer using the `--provider` flag from here https://github.com/fluxcd/flux2/blob/main/cmd/flux/create_source_oci.go#L74

Reply:
Hi and thanks for the review 👋 

I've pushed a new change that moved that logic to the OCI client, let me know!

Regarding the change in the method arguments, AFAIK, adding variadic arguments does not break the existing usage of the function: https://go.dev/play/p/UTWaaDYbwv-

Reply:
What do you think or replacing --auto-login with --provider?

Reply:
I don't mind either, the auto-login allowed  depending on the end-user flow and the UX of Flux (that I'm still discovering)

Code-wise, in `pkg/oci` that would mean adding a switch over the provider value, then initiating a client for each supported registries, then calling the login function for each of them which would look like this code already present: https://github.com/fluxcd/pkg/blob/main/oci/auth/login/login.go#L97 so I'm not sure if it's worth to duplicate this feature/logic.

```go
func (m *Manager) Login(ctx context.Context, image string, ref name.Reference, opts ProviderOptions) (authn.Authenticator, error) {
	switch ImageRegistryProvider(image, ref) {
	case oci.ProviderAWS:
		return m.ecr.Login(ctx, opts.AwsAutoLogin, image)
	case oci.ProviderGCP:
		return m.gcr.Login(ctx, opts.GcpAutoLogin, image, ref)
	case oci.ProviderAzure:
		return m.acr.Login(ctx, opts.AzureAutoLogin, image, ref)
	}
	return nil, nil
}
```

Reply:
It feels really strange to ask for a specific provide everywhere in Flux except for this command.

Reply:
I've noticed that the provider enum is not the same from the CLI and in the pkg/oci:

* https://github.com/fluxcd/flux2/blob/08401f62b2ad7b4573cb85847a56c35288aa530f/internal/flags/source_oci_provider.go#L28
* https://github.com/fluxcd/source-controller/blob/main/api/v1beta2/ocirepository_types.go#L27

vs

https://github.com/fluxcd/pkg/blob/770524e58984e6165ae7e2fc92da491ec4b1b7c1/oci/constants.go#L23

Which makes interop a bit tricky, I keep on digging 🕵️

Reply:
I've updated the PR to add a mapping between the two types with additional logic in pkg/oci 👍

Reply:
Looks great! Thank you. Can you please add the 2 flags to tag|pull|list artifact(s)